src/modules/wallet/components/request/request.js

```diff
@@ -32,61 +61,41 @@ const Account = () => {
   );
 };
 
-class Request extends React.Component {
-  constructor(props) {
-    super();
-
-    this.state = {
-      shareLink: `lisk://wallet/send?recipient=${props.address}`,
-      fields: {
-        amount: {
-          error: false,
-          value: '',
-          loading: false,
-          feedback: '',
-        },
-        reference: {
-          error: false,
-          value: '',
-          loading: false,
-          feedback: '',
-        },
-        token: {
-          error: false,
-          value: '',
-          loading: false,
-          feedback: '',
-        },
-        recipientApplication: {
-          error: false,
-          value: '',
-          loading: false,
-          feedback: '',
-        },
-      },
-    };
-
-    this.timeout = {
-      amount: null,
-      reference: null,
-    };
-
-    this.handleFieldChange = this.handleFieldChange.bind(this);
-    this.updateShareLink = this.updateShareLink.bind(this);
-    this.onSelectReceipentChain = this.onSelectReceipentChain.bind(this);
-    this.onSelectToken = this.onSelectToken.bind(this);
-    this.onRemoveMessageField = this.onRemoveMessageField.bind(this);
-  }
-
-  /* istanbul ignore next */
-  componentWillUnmount() {
-    clearTimeout(this.timeout.amount);
-    clearTimeout(this.timeout.reference);
-  }
+const Request = () => {
+  const [
+    {
+      metadata: { address },
+    },
+  ] = useCurrentAccount();
+
+  const { t } = useTranslation();
+  const [state, dispatch] = useReducer(
+    (stateData, value) => ({ ...stateData, ...value }),
+    requestInitState
+  );
+  const timeout = useRef({
+    amount: null,
+    reference: null,
+  });
+
+  const { applications } = useApplicationExploreAndMetaData();
+  const { data: tokens } = useTransferableTokens(state.recipientChain.value);
```

Comment:
Now when we use this as a hook this will produce RQ cache side effect when typing amount before selecting an application. RQ will cache a wrong client in `useTransferableTokens`... So we need to extend to extend our `useTokensSupported` hook to include an enabled option and use like this 

```
  } = useTokensSupported({
    client: client.current,
    options: { enabled: !!application?.serviceURLs?.[0] },
  });
  ```

Reply:
![image](https://user-images.githubusercontent.com/8784876/214052300-c19fd3d4-db04-400e-b5a4-0be83c12cde8.png)

To prevent this race condition with RQ